marketing/views.py

```diff
@@ -15,9 +18,53 @@ class QuoteListView(generics.ListAPIView):
 class QuoteCreateView(generics.CreateAPIView):
     serializer_class = QuoteSerializer
 
+    def create(self, request, *args, **kwargs):
+        program_id = request.headers.get('Program-ID')
+        quote_serializer = self.get_serializer(data=request.data)
+        quote_serializer.is_valid(raise_exception=True)
+        quote = quote_serializer.save(author=request.user)
+
+        try:
+            program = Program.objects.get(id=program_id)
+            program.quotes.add(quote)
+            return Response({"message": "Quote attached successfully"})
+        except Program.DoesNotExist:
+            return Response({"message": "Program not found"})
```

Comment:
When we return an error, we should modify the return code to something that makes sense. Like with this exception, we should probably be returning an HTTP status of 400.